### ISSUE TITLE
Rule 0302 should not consider special symbols

### ISSUE THREAD

**Issue by hajdum** [MarketSquare/robotframework-robocop#226]
**Describe the bug**
Rule 0302 should not consider special symbols such as - , = ....

**To Reproduce**
1. Create keyword with name:   THIS-IS-MY-KEYWORD
2. run robot
3. observe

**Expected behavior**
0302 rule notification is not present

**Comment by bhirsz** [comment]
Valid point. There is also this problem that Robocop when analyzing the keyword does not know if the keyword have embedded variables. For example:

```
*** Keywords ***
My Keyword With ${var} Variable    # this is handled as we recognize we have variable here
    No Operation

Other Keyword
    My Keyword With small Variable  # this will throw warning
```

This could be somehow connected. But for rule 0302 I could try to strip any non letter character and then run check. Ie
THIS-IS-MY-KEYWORD -> THIS IS MY KEYWORD

Of course some characters are not desired in keyword name but it should not raise warning with 0302 (keyword not capitalized) but with other rule.

**Comment by bhirsz** [comment]
Example fix at fix/not_capitalized_special_chars

You can checkout this branch, install it from source code (pip install -e . if in roboframework-robocop directory) and use it for test. 

I will spend some extra time thinking about best algorithm for this rule before merging anything
